src/eth/tests.rs

```diff
@@ -113,6 +118,15 @@ fn sample1_decode() {
 	assert_eq!(v2, true);
 }
 
+#[test]
+fn fixed_size_decode() {
+	let payload: &[u8] = &[
+		0x45, 0x01, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x00, 0x00,
+	];
+	let array = single_decode::<[u8; 30]>(payload);
+	assert_eq!(array, [0x45, 0x01, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08]);
```

Comment:
better use 30 non-zero values, not 3

Reply:
the last byte is non-zero

Reply:
cool, maybe make it two lines so everyone would see